src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+        ProductResponseDto productResponseDto = response.getBody();
+
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+
+        assertEquals(productRequestDto.getName(), Objects.requireNonNull(productResponseDto).getName());
+        assertEquals(productRequestDto.getResourcesContent().get(0).getId(), productResponseDto.getResourcesContent().get(0).getResource().getId());
+    }
+
+    @Test
+    void getProductSuccessfully() {
+
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.getForEntity(getProductUrl(Objects.requireNonNull(productResponseDto).getId()), ProductResponseDto.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void getAllProductsSuccessfully() {
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<List<ProductResponseDto>> response =
+                this.testRestTemplate.exchange(
+                        getBaseProductUrl(), HttpMethod.GET, null, new ParameterizedTypeReference<>() {
+                        });
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void deleteProductSuccessfully() {
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<HttpStatus> response =
+                this.testRestTemplate.exchange(
+                        getProductUrl(Objects.requireNonNull(productResponseDto).getId()),
+                        HttpMethod.DELETE,
+                        null,
+                        HttpStatus.class);
+
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+    }
+
+
+    @Nullable
+    private ProductResponseDto getProductResponseDto(ProductRequestDto productRequestDto) {
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+
+        return response.getBody();
+    }
+
+    @NotNull
+    private static ProductRequestDto getProductRequestDto(ResourcesInUserResponseDto resourcesInUser, User user) {
+
+        List<ResourceQuantityRequestDto> listOfResourcesInProduct = getResourceInProductRequestDtos(resourcesInUser);
+
+        ProductRequestDto productRequestDto = new ProductRequestDto();
+        productRequestDto.setName("TestProduct");
+        productRequestDto.setAuthors(List.of("TestAuthors"));
+        productRequestDto.setDescription("Test");
+        productRequestDto.setOwnerId(user.getId());
+        productRequestDto.setSalePrice(50);
+        productRequestDto.setResourcesContent(listOfResourcesInProduct);
+
+        return productRequestDto;
+    }
```

Comment:
Should we move this to the helper as well?

Reply:
Done!